sqlglotrs/src/tokenizer.rs

```diff
@@ -399,6 +399,19 @@ impl<'a> TokenizerState<'a> {
             } else if *token_type == self.token_types.bit_string {
                 (Some(2), *token_type, end.clone())
             } else if *token_type == self.token_types.heredoc_string {
+                if self.settings.heredoc_tag_is_identifier
+                    && !(self.peek_char.is_alphabetic() || self.peek_char == '_')
```

Comment:
```
|| self.peek_char == '_'
```
this is not in the Python version either.

Reply:
That's right, the Python version uses `isidentifier()` which aims to replicate this logic. I searched tokenizer.rs and it seems like the other `isidentifier()` call is mirrored like what I did here, inside `scan_number()`. Should we factor this out in a helper with a similar name? Is there no rust equivalent for alpha + `_` checking?

Reply:
> Should we factor this out in a helper with a similar name? Is there no rust equivalent for alpha + _ checking?

ah, makes sense. No, I didn't find one